### ISSUE TITLE
Fix history command

### ISSUE THREAD

**Issue by rgardner** [rgardner/bsh#10]
### positive indices

```
$ echo test
$ !1
Segmentation fault: 11
```

### negative indices

```
$ echo test
$ !-1
tes
$ # should be test
$ echo test2
$ !-1
test
$ # should be test2
$ !-5
Segmentation fault: 11
```

**Comment by rgardner** [comment]
_facepalm_ `strlcpy` takes a size, but returns a **length**. I thought it returned a size. `strlcpy` always null-terminates, but it was always missing a character.

**Comment by rgardner** [comment]
fixed in @ecd907b7c6692e83bcfec20ef13c1bcdd9cd22b2 and @ 9e50e80191a72f0436fc87018eaefce8cdcf9a59
